core_brain/kpi.py

```
@@ -47,6 +47,156 @@ def _wilson_ci(successes: int, n: int, z: float = 1.96) -> Optional[dict[str, fl
+    for d in deltas:
+        n = required_sample_size(sigma, d, z_alpha, z_beta)
+        hw: float | None = None
+        if n is not None and n > 0:
+            ci = _wilson_ci(n // 2, n)  # 50% win rate = worst-case width
+            if ci is not None:
+                hw = (ci["upper"] - ci["lower"]) / 2.0
+        rows.append({"delta": d, "n_required": n, "wilson_half_width": hw})
+    return rows
+
+
+def gate_definition(cfg: "MakerConfig | None" = None) -> dict[str, Any]:
```

Ruff (0.16.2)

[warning] Remove quotes from type annotation

Remove quotes

(UP037)